### ISSUE TITLE
MSFS add-on airports from the community folder are completely ignored on loading scenery 

### ISSUE THREAD

**Issue by Emma-Peel** [albar965/littlenavmap#967]
- Little Navmap Version 2.8.7
- Windows 11 22H2
- Navigraph AIRAC Cycle 2212 rev.1
- MSFS Version 1.29.30.0

Starting with Little Navmap Version 2.8.7, loading the scenery from MSFS does no longer work for add-on airports. Since this is such an important feature I could not imagine it could be broken generally, but all investigations on my side did not lead to any possible problem.

Problem:
- Scenery Import for MSFS runs without errors but completly ignores any add-on scenery.
- After each run the following error message is displayed (but is not true):

  ![grafik](https://user-images.githubusercontent.com/91339262/209323787-ee8b0ae1-6dc8-411e-bd0a-47978d36ea54.png)

- I assume that the current installed navdata is detected from the folder Packages\Community\navigraph-navdata\ which is available on my system. The Navigraph installer itsself reports the navadata as installed as well.
- It worked well with the previous version of Little Navmap.

Some diagnostic I did so far:
- Deleting the existing databases and retry: No change.
- I took a look into the little_navmap_msfs.sqlite and found only BGL files from Packages\Official\OneStore\fs-base-genericairports\ and Packages\Official\OneStore\fs-base-nav\
- A did another run while monitoring file access with Sysinternal Process Monitor: The only action regarding the Packages\Community folder was an enumeration of the sub folders of that folder in a relativ early stage of processing. From then on, no further attempt to access Packages\Community or a subfolder has been recording. 
- I did also did not find anything suspicious in the log file: [abarthel-little_navmap.log](https://github.com/albar965/littlenavmap/files/10294624/abarthel-little_navmap.log)

Any help to resolve this issue is very appreciated. If there is anythng I can do on my side to support further investigations, let me know.

**Comment by albar965** [comment]
> * Scenery Import for MSFS runs without errors but completly ignores any add-on scenery.

Do you use an add-on manager tool? LNM should normally be able to deal with all kind of shortcuts and links that are used by these tools. In any case make sure that the add-on folders are not nested within other folders. Add-on folders containing manifest.json and layout.json have to reside on the top level of Community.
For example like this:
![grafik](https://user-images.githubusercontent.com/15254956/209329577-67fd344a-af71-46ff-944f-68a3ec66e468.png)

> * After each run the following error message is displayed (but is not true):

This is not an error message but more a hint to switch to the right scenery library mode. Clicking yes will change the settings in the scenery library menu to the recommended mode according to https://www.littlenavmap.org/manuals/littlenavmap/release/2.8/en/SCENERY.html#microsoft-flight-simulator-2020-airports-and-navdata.

You mean that you have the Navigraph update installed? Then ignore it and consider the link above.
This might be an error in LNM. Will check.

```
Checking MSFS path from "C:\\Users\\juergen\\AppData\\Local\\Packages\\Microsoft.FlightSimulator_8wekyb3d8bbwe\\LocalCache\\UserCfg.opt"
Found MSFS base path "C:/FS/MSFS/Packages"
```
I suppose "C:/FS/MSFS/Packages" is the correct path and this is no Steam installation.

```
Found MSFS community path "C:/FS/MSFS/Packages\\Community"
Found MSFS official path "C:/FS/MSFS/Packages\\Official\\OneStore"
```
Are these the correct paths?

```
File "C:/manifest.json" does not exist
File "C:/layout.json" does not exist
```
No idea why this happens. Maybe an empty path but certainly something is wrong there.

What to try:
1. Try to put add-ons directly into Community if you use links or shortcuts.
2. Remove add-ons out of nested folders as explained above.
4. make sure that there are no excluded folders in LNM options on page `Scenery Library Database`
3. Click `Reset Paths` in the load scenery library dialog and reload.

**Comment by Emma-Peel** [comment]
> Do you use an add-on manager tool? LNM should normally be able to deal with all kind of shortcuts and links that are used by these tools. In any case make sure that the add-on folders are not nested within other folders. Add-on folders containing manifest.json and layout.json have to reside on the top level of Community. 

Yes, I do and there are many symbolik links in the community folder. But this was never a problem with previous version I do regular LVM updates after changing scenery.

> This is not an error message but more a hint to switch to the right scenery library mode. Clicking yes will change the settings in the scenery library menu to the recommended mode according to https://www.littlenavmap.org/manuals/littlenavmap/release/2.8/en/SCENERY.html#microsoft-flight-simulator-2020-airports-and-navdata.
> 
> You mean that you have the Navigraph update installed? Then ignore it and consider the link above. This might be an error in LNM. Will check.

Yes, the Navigraph update is installed and the warning is not applicable.  

> Checking MSFS path from "C:\\Users\\juergen\\AppData\\Local\\Packages\\Microsoft.FlightSimulator_8wekyb3d8bbwe\\LocalCache\\UserCfg.opt"
> Found MSFS base path "C:/FS/MSFS/Packages"
> I suppose "C:/FS/MSFS/Packages" is the correct path and this is no Steam installation.
> Found MSFS community path "C:/FS/MSFS/Packages\\Community"
> Found MSFS official path "C:/FS/MSFS/Packages\\Official\\OneStore"
> 
> Are these the correct paths?

Yes, that is all correct.

> File "C:/manifest.json" does not exist
> File "C:/layout.json" does not exist
> ```
> 
> No idea why this happens. Maybe an empty path but certainly something is wrong there.

Me neither. Two additions: 
- I have noticed the mixed use of forward and backward slashes in the paths logged above. May this lead to problems accessing the json-files?
- I also would like to mention that the "FS" of C:/FS/MSFS/Packages is not a regular folder but a mapped drive. I have never ever seen any issues with any software using this configuration but you never now...

> What to try:
> 
>     1. Try to put add-ons directly into Community if you use links or shortcuts.
>     2. Remove add-ons out of nested folders as explained above.

I removed all sub folders from the Community folder except one physical folder containing one airport. Checked that there are no nested folders and the json files exist in the expected location and loaded the scenery again. Result:

- Even that single airport has not been loaded. That fits to the results of the mentioned process monitor monitoring as it claims, that LNM does not try to make any file access below the Community root folder C:\FS\MSFS\Packages\Community  
- There are still many ""C:/manifest.json" does not exist" warnings in the log file.
- While having a look to the Packages folder I recognized that the issue does not only apply the community folder but as well to the handcrafted airport models that come with MSFS like EDDS. I am pretty sure that is was markes as add-on in the previous version of LNM.

>     3. make sure that there are no excluded folders in LNM options on page `Scenery Library Database`

This has always been empty.

>     4. Click `Reset Paths` in the load scenery library dialog and reload.

I tried this before with no success.

**Comment by albar965** [comment]
Thank you for your answers. 

There were a lot of changes in the 2.8 release to support symbolic links, Windows shortcuts and junctions as well as macOS aliases. Maybe I buried some gremlins in there while adding support for the links.

I have to check if LNM does not detect the Navigraph update. Maybe Navigraph changed some data in the manifest I use to detect it.

The mix of forward and backward slashes is no issue. LNM can deal with this internally. I only try to provide clean paths on the user interface but in the log it looks messy.

The mapped drive should not cause any issues.

Furthermore LNM detects all as add-on which is not stored in fs-base and the other stock folders.

Give me some time. I need to have a very close look at the log file and experiment with mapped drives.

**Comment by Emma-Peel** [comment]
OK, thank you so far. I am now back to Version 2.6.19 and everything works fine again. I am curious to hear you findings, but for the meantime have happy holidays!

**Comment by albar965** [comment]
BTW: LNM accesses all folders and collects BGL files from the layout.json to have an estimate for the progress bar. This explains the strange file access characteristics.

Wish you Happy Holidays too!

**Comment by albar965** [comment]
Fixed with albar965/atools@0d25d0d652f99eaf61dfeee6c20e50f0ec409f79
At least it is working with a mapped drive in my test VM.

**Comment by Emma-Peel** [comment]
I can confirm that the issue is solved with LNM 2.8.8 - thank you very much!

**Comment by albar965** [comment]
> I can confirm that the issue is solved with LNM 2.8.8 - thank you very much!

Thank you a lot for the feedback. Glad to hear that it is not only fixed on my test system. :slightly_smiling_face:
